### PR TITLE
[Improve][Connector-V2][File] Support split file based on batch size

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

support batch size for file sink

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Review by PraveenNanda124 — COMMENTED** [review]
Good work

File: docs/en/connector-v2/sink/FtpFile.md [hunk 1]
```diff
@@ -127,6 +128,10 @@ Please note that, If `is_enable_transaction` is `true`, we will auto add `${tran
 
 Only support `true` now.
 
+### batch_size [int]
+
+Write a file for every number of pieces of data
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/FtpFile.md]: We need let user know `batch_size` is the maximum number of rows in a file. For SeaTunnel Engine, The number of lines in the file is determined by `batch_size` and `checkpoint.interval` jointly decide. If the value of `checkpoint.interval` is large enough, sink writer will write rows in a file until the rows in the file large than `batch_size`. If `checkpoint.interval` is small, the sink writer will create a new file when a new checkpoint trigger.

**Comment by EricJoy2048** [comment]
You can fix the dead link ci failed by merge code from dev.

**Comment by EricJoy2048** [comment]
Please fix CI error.

File: README.md [hunk 2]
```diff
@@ -86,13 +86,13 @@ Download address for run-directly software package : https://seatunnel.apache.or
...
-https://seatunnel.apache.org/docs/category/start-v2
+https://seatunnel.apache.org/docs/category/source-v2
 
 **Spark**
-https://seatunnel.apache.org/docs/category/start-v2
+https://seatunnel.apache.org/docs/category/source-v2
 
 **Flink**
-https://seatunnel.apache.org/docs/category/start-v2
+https://seatunnel.apache.org/docs/category/source-v2
```
Comment (EricJoy2048) [inline README.md]: we need add `quick start` link.

Comment (TyrantLucifer) [reply README.md]: Resolved.

**Review by EricJoy2048 — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -86,13 +86,13 @@ Download address for run-directly software package : https://seatunnel.apache.or
 ## Quick start
 
 **SeaTunnel Engine**
-https://seatunnel.apache.org/docs/category/start-v2
+https://seatunnel.apache.org/docs/start-v2/locally/quick-start-seatunnel-engine
```
Comment (TaoZex) [inline README.md]: The dead link have been fixed. Please remove it.

**Review by EricJoy2048 — APPROVED** [review]

**Review by TaoZex — APPROVED** [review]
+1

**Review by wuchunfu — APPROVED** [review]

**Pull request merged**